validate.js

```diff
@@ -1114,6 +1114,46 @@
       if (!PATTERN.exec(value)) {
         return message;
       }
+    },
+
+    // A type validator used to enforce generic types on data.
+    // wraps extant isThing helpers
+    type: function(value, options) {
+      // a/an selector to produce proper grammer in error messages
+      function aOrAn(nextWord) {
+        var vowels = "aeiou";
+        if (( typeof nextWord !== "string" ) || !nextWord.length) {
+          return "a";
+        }
+        return (vowels.indexOf(nextWord[0]) === -1) ? "a" : "an";
+      }
+      var typeHelperMap = {
```

Comment:
The `isBoolean` and `isHash` utility functions were recently added to the library and would be easy to add here.
```hash: v.isHash,
boolean: v.isBoolean```